roofit/xroofit/src/xRooFit.cxx

```diff
@@ -1124,9 +1121,12 @@ int xRooFit::minos(RooAbsReal &nll, const RooFitResult &ufit, const char *parNam
    fitConfig.SetMinosErrors(mErrs);
    par->setConstant(isConst);
 
-   const_cast<RooFitResult &>(ufit)._statusHistory.push_back(
-      std::make_pair(std::string(TString::Format("xMINOS_%s", parName)), status));
-   const_cast<RooFitResult &>(ufit)._status += status;
+   std::vector<std::pair<std::string, int>> statusHistory;
+   for (int i = 0; i < ufit.numStatusHistory(); i++)
+      statusHistory.emplace_back(ufit.statusLabelHistory(i), ufit.statusCodeHistory(i));
```

Comment:
```suggestion
   for (unsigned int i = 0; i < ufit.numStatusHistory(); i++) {
      statusHistory.emplace_back(ufit.statusLabelHistory(i), ufit.statusCodeHistory(i));
   }
```
To avoid type mismatch warnings. Also, just a general thing for the next time, we try to always use braces even for one statement. That makes the code easier to read in my opinion, but feel free to ignore this comment if you disagree